### PR TITLE
New github action for gathering metrics

### PR DESCRIPTION
**Pull request by wojteknowacki**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [x] 🔩 Maintenance
- [x] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

## [Optional] Description
  🔑 Details of the action [used](https://github.com/marketplace/actions/issue-metrics)

### CODE REVIEW

File: .github/workflows/repo-metrics.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+          #Set an environment variable with the date range
+          echo "$first_day..$last_day"
+          echo "last_month=$first_day..$last_day" >> "$GITHUB_ENV"
+
+      - name: Run issue-metrics tool
+        uses: github/issue-metrics@v3
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SEARCH_QUERY: 'saleor-dashboard:saleor/repo is:issue created:${{ env.last_month }} -reason:"not planned"'
```
Comment (wojteknowacki) [inline .github/workflows/repo-metrics.yml:33]: Not entirely sure about about this part `saleor-dashboard:saleor/repo` but need to merge to check it out

**Review by stmpn — APPROVED** [review]
Looks good to me, but I have few suggestions:

* We could consider configuring cron schedule to automatically run the report generation on first day of the month.
* The issue created for "publishing" monthly report should contain the MM-YYYY date in the title
* If you parametrize the repo name that is used in SEARCH_QUERY, this workflow can be implemented for other repos basically out of the box (not sure if we want that though).

**Comment by wojteknowacki** [comment]
@stmpn Thx 👍 I agree with all you wrote but for now, I treat it as POC. I fixed the date in the file name but the rest we can update if this is sth we want to keep.

File: .github/workflows/repo-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          report_filename="issue_metrics_$(date +'%Y-%m').md"
+          echo "This is the monthly issue metrics report." > "$report_filename"
+          echo "Add your metrics data here." >> "$report_filename"
+          echo "report_filename=$report_filename" >> "$GITHUB_ENV"
+
+      - name: Create issue
+        uses: peter-evans/create-issue-from-file@v5
+        with:
+          title: Monthly issue metrics report
```
Comment (stmpn) [inline .github/workflows/repo-metrics.yml:45]:
```suggestion
          echo "report_filename=$report_filename" >> "$GITHUB_ENV"
          # Set an environment variable with the tag which we append to the title of GH issue.
          echo "last_month_tag=$(date -d "last month" +%Y-%m)" >> "$GITHUB_ENV"

      - name: Create issue
        uses: peter-evans/create-issue-from-file@v5
        with:
          title: Monthly issue metrics report for ${{ env.last_month_tag }}
```

**Review by stmpn — DISMISSED** [review]

**Review by stmpn — APPROVED** [review]

File: .github/workflows/repo-metrics.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          report_filename="issue_metrics_$(date +'%Y-%m').md"
+          echo "This is the monthly issue metrics report." > "$report_filename"
+          echo "Add your metrics data here." >> "$report_filename"
+          echo "report_filename=$report_filename" >> "$GITHUB_ENV"
+          # Set an environment variable with the tag which we append to the title of GH issue.
+          echo "last_month_tag=$(date -d "last month" +%Y-%m)" >> "$GITHUB_ENV"
+
+      - name: Create issue
+        uses: peter-evans/create-issue-from-file@v5
```
Comment (NyanKiyoshi) [inline .github/workflows/repo-metrics.yml:43]: Freeze the 3rd party action:
```suggestion
        uses: peter-evans/create-issue-from-file@24452a72d85239eacf1468b0f1982a9f3fec4c94 # v5.0.0
```

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Review by stmpn — APPROVED** [review]

**Pull request merged**